kore/src/Kore/Syntax/Json.hs

```diff
@@ -94,11 +97,40 @@ prettyJsonOpts =
 ------------------------------------------------------------
 -- convenience converters
 
-fromTermLike :: TermLike.TermLike VariableName -> KoreJson
+fromTermLike :: TermLike VariableName -> KoreJson
 fromTermLike =
     addHeader
         . fromPattern
         . from @_ @(Kore.Pattern _ (TermLike.TermAttributes VariableName))
 
 fromPredicate :: Kore.Sort -> Predicate VariableName -> KoreJson
 fromPredicate s = fromTermLike . Predicate.fromPredicate s
+
+{- | represent a @'Substitution'@ as a conjunction of equalities, so
+'[t1 / X][t2 / Y]'becomes '#And ( X #Equals t1, #And ( Y #Equals t2, ... ))'.
+The result sort is fixed to a sort variable.
+-}
+fromSubstitution :: Substitution VariableName -> Maybe KoreJson
+fromSubstitution subst
+    | Substitution.null subst = Nothing
+    | otherwise =
+        Just
+            . fromTermLike
+            . foldl1 TermLike.mkAnd
```

Comment:
this will technically produce

```
[t1 / x1][t2 / x2]..[tn / xn] ~> #And ( ... And (x1 #Equals t1, x2 #Equals t2) ..., xn #Equals tn )
```
... i think... never remember which way folds go...

Reply:
You are right, the code does not match the description. I'll fix the description then.

Reply:
thanks :)